internal/pkg/agent/application/actions/handlers/handler_action_policy_change.go

```diff
@@ -220,25 +224,80 @@ func (h *PolicyChangeHandler) handleFleetServerHosts(ctx context.Context, c *con
 	return nil
 }
 
-func clientEqual(k1 remote.Config, k2 remote.Config) bool {
-	if k1.Protocol != k2.Protocol {
+// applyConfigWithPrecedence applies Proxy and TLS configs, but ignores empty or
+// absent ones.
+// That way a proxy or TLS config set during install/enroll using cli flags
+// won't be overridden by an absent or empty proxy from fleet-server.
+// However, if there is a proxy or TLS config sent by fleet-server, it'll take
+// precedence. Therefore, it's not possible to remove a proxy or TLS config once
+// it's set.
+func (h *PolicyChangeHandler) applyConfigWithPrecedence(cfg remote.Config) {
+	defaultcfg := configuration.DefaultFleetAgentConfig()
+
+	if cfg.Protocol != defaultcfg.Client.Protocol ||
+		cfg.Host != defaultcfg.Client.Host ||
+		!slices.Equal(cfg.Hosts, defaultcfg.Client.Hosts) ||
+		cfg.Path != defaultcfg.Client.Path {
+		h.config.Fleet.Client.Protocol = cfg.Protocol
```

Comment:
Why do we overwrite very parameter if only one doesn't match the default?

Reply:
There is a built in assumption that Fleet will never change this unintentionally.

Do not assume Fleet will always perfectly conform to your expectations. Bugs can exist on both sides of the connection.

If this happened and caused settings not apply, how would we tell and debug it?